pkgs/applications/networking/cluster/jx/default.nix

```diff
@@ -1,49 +1,49 @@
-{ stdenv, buildGoModule, fetchFromGitHub, lib, installShellFiles }:
+{ stdenv, buildGoModule, fetchFromGitHub, lib, nix-update-script, go }:
 
 buildGoModule rec {
   pname = "jx";
-  version = "2.1.155";
+  version = "3.10.144";
 
   src = fetchFromGitHub {
     owner = "jenkins-x";
     repo = "jx";
     rev = "v${version}";
-    sha256 = "sha256-kwcmZSOA26XuSgNSHitGaMohalnLobabXf4z3ybSJtk=";
+    sha256 = "sha256-1/gW9p2M0ZW3H9rM7JNdsh19A4yIO8SCE8u6UpB4ftI=";
   };
 
-  vendorHash = "sha256-ZtcCBXcJXX9ThzY6T0MhNfDDzRC9PYzRB1VyS4LLXLs=";
+  vendorHash = "sha256-AIaZVkWdNj1Vsrv2k4B5lLE0lOFuiTD7lwS/DikmC14=";
 
-  doCheck = false;
+  subPackages = [ "cmd" ];
 
-  subPackages = [ "cmd/jx" ];
-
-  nativeBuildInputs = [ installShellFiles ];
+  CGO_ENABLED = 0;
```

Comment:
Is there a reason for the override? From a quick test package seems to build fine without it.

Reply:
It is good practice to disable cgo where possible. Just nicer to have a static build. Should be part of buildGoModule, too, but backwards compat...
Upstream also disables cgo for their release builds: https://github.com/jenkins-x/jx/blob/c297bef1a209dcba111006dfdb68c206912994e2/.goreleaser.yml#L3

Reply:
Upstream does because it is kind of the best way they have to produce something that will in most cases. nixpkgs is in a different situation since we are in control of what the binary will use.

Anyway this probably a discussion that should occur outside of a specific PR doing the upgrade of a package and I have no strong opinion. Anyway we are still in a better place with this package upgraded :+1: